pkg/policy/input_resolver.go

```diff
@@ -39,8 +40,13 @@ func (r *inputResolver) resolve(ctx context.Context, query ResourcesQuery) (Reso
 	ret := ResourcesResult{ScopeFound: true}
 	if resources, ok := r.input.Resources[query.ResourceType]; ok {
 		ret.ScopeFound = true
-		for _, resource := range resources {
-			ret.Resources = append(ret.Resources, resource)
+		keys := []string{}
+		for k := range resources {
+			keys = append(keys, k)
+		}
```

Comment:
It's easy to preallocate the slice (or rather, the underlying storage array) here already:
```suggestion
		keys := make([]string{}, 0, len(resources))
		for k := range resources {
			keys = append(keys, k)
		}
```
The `make(..., 0, n)` creates a new slice with a 0-length but n-capacity. When using `append`, it will thus not need to grow the slice, ergo less allocations.